use-cases/bed-allocation/src/main/java/org/acme/bedallocation/rest/DemoDataGenerator.java

```diff
@@ -333,4 +331,51 @@ private <T, L> void applyRandomValue(int count, List<T> values, L secondParam, P
             }
         }
     }
+
+    private record Pair<Key_, Value_>(Key_ key, Value_ value) {
+        public Pair(Key_ key, Value_ value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public Key_ key() {
+            return this.key;
+        }
+
+        public Value_ value() {
+            return this.value;
+        }
+    }
+
+    private final class MutableReference<Value_> {
+        private Value_ value;
+
+        public MutableReference(Value_ value) {
+            this.value = value;
+        }
+
+        public Value_ getValue() {
+            return this.value;
+        }
+
+        public void setValue(Value_ value) {
+            this.value = value;
+        }
+
+        public boolean equals(Object o) {
+            if (o instanceof MutableReference<?> other) {
+                return this.value.equals(other.value);
+            } else {
+                return false;
+            }
+        }
+
+        public int hashCode() {
+            return Objects.hash(this.value);
+        }
+
+        public String toString() {
+            return this.value.toString();
+        }
+    }
```

Comment:
Considering that this is a quickstart, let's refactor the code so that mutable references aren't necessary. Every time you have to ask yourself "why is this here?" and the answer is not directly related to the idea you want to show the users, remove the thing.